src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_SwitchExpression.cs

```diff
@@ -51,11 +57,26 @@ private BoundExpression LowerSwitchExpression(BoundConvertedSwitchExpression nod
                     node.DecisionDag, loweredSwitchGoverningExpression, result, out BoundExpression savedInputExpression);
                 Debug.Assert(savedInputExpression != null);
 
+                object restorePointForEnclosingStatement = new object();
```

Comment:
nit: do we use `object` instances as identifiers elsewhere? Seems like it would make it difficult to debug/maintain. 
Would using numbers as identifiers instead cause problems/bottlenecks?

Reply:
Yes, we do use `object` instances as identifiers in this layer (e.g. for labels).